packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -92,10 +92,14 @@ abstract class TransitionRoute extends OverlayRoute {
 
   void didPop(dynamic result) {
     _result = result;
-    if (completer != null)
-      _performance.reverse().then((_) { completer.complete(_result); });
+    void finish() {
```

Comment:
I think this would be clearer without this nested function.

Its body can just be `completer?.complete(result);`.

That can probably just be inlined in the two places you call it.

Also, if I'm reading the code right, you don't need to explicitly do the reverse().then(finish) logic, because the `_handleStatusChanged` method already does that. No?